tools/patch-zephyr-sdk-toolchain-download.py

```
@@ -0,0 +1,30 @@
+
+Some macOS hosts (observed on a self-hosted CI runner) fail every invocation
+of the SDK's bundled hosttools wget when downloading a GNU toolchain archive,
+even against a freshly extracted SDK, while a plain curl to the same URL
+succeeds. This appends a curl fallback to the wget invocation in setup.sh so
+a wget failure doesn't abort the install.
+"""
+
+import sys
+
+path = sys.argv[1]
+content = open(path).read()
```

Ruff (0.15.21)

[warning] Use a context manager for opening files

(SIM115)